dsalglib/include/Number_Theory_Algorithms/fundamental_functions.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Created by yadvendra20 on 3/12/2020
+*/
+
+#ifndef DSALGLIB_NUMTHEORY_FF_H
+#define DSALGLIB_NUMTHEORY_FF_H
+
+#include "../alginc.h"
+
+namespace dsa {
+	
+	/*
+	 * The binary exponentiation method computes 
+	 * powers directed by the exponent bits, one 
+	 * at a time. Bits of value 0 require a squaring; 
+	 * bits of value 1 require a squaring and a multiplication.
+	 * 
+	 * More about binary exponentiantion: https://cp-algorithms.com/algebra/binary-exp.html
+	*/
+	
+	long long binaryExponentiation(long long base, long long exponent) {
```

Comment:
Consider effects of bits when dealing with small number but not similar data type